aws-wildfly/src/main/java/sunstone/aws/impl/AwsWFArchiveDeployer.java

```diff
@@ -57,11 +63,30 @@ static void undeployFromEc2Instance(String deploymentName, Identification resour
     }
 
     static void deployToEc2Instance(String deploymentName, Identification resourceIdentification, WildFly wildFly, InputStream is, AwsSunstoneStore store) throws SunstoneException {
-        try (OnlineManagementClient client = AwsWFIdentifiableSunstoneResourceUtils.resolveOnlineManagementClient(resourceIdentification, wildFly, store)){
-            client.apply(new Deploy.Builder(is, deploymentName, true).build());
-        } catch (CommandFailedException e) {
-            throw new SunstoneException(e);
-        } catch (IOException e) {
+        Deploy.Builder builder = new Deploy.Builder(is, deploymentName, true);
+        DomainMode domainMode = wildFly.domain();
+        //no further configuration needed for standalone mode,
+        //in domain mode, we need to specify server groups
+        if (domainMode != null) {
+            if (wildFly.domain().serverGroups() == null) {
+                throw new RuntimeException(WildFlyConfig.DOMAIN_SERVER_GROUPS + " is not set");
+            }
+            String serverGroupsParam = String.join(",", wildFly.domain().serverGroups());
+
+            //groups may not be set -> deploy to all groups
+            Optional<String> serverGroups = isExpression(serverGroupsParam) ? SunstoneConfig.resolveOptionalExpression(serverGroupsParam, String.class) : Optional.of(serverGroupsParam);
+            if (serverGroups.isPresent()) {
+                Arrays.stream(serverGroups.get().split(","))
```

Comment:
Could we benefit from SR Config implementation to get the list? https://download.eclipse.org/microprofile/microprofile-config-3.0/microprofile-config-spec-3.0.html#configexamples

WDYT?